app/main.py

```diff
@@ -1,214 +1,28 @@
-
-        if "hp" in mordred["potion"]["effect"]:
-            mordred["hp"] += mordred["potion"]["effect"]["hp"]
-
-    # red_knight
-    red_knight = knightsConfig["red_knight"]
-
-    # apply armour
-    red_knight["protection"] = 0
-    for a in red_knight["armour"]:
-        red_knight["protection"] += a["protection"]
-
-    # apply weapon
-    red_knight["power"] += red_knight["weapon"]["power"]
-
-    # apply potion if exist
-    if red_knight["potion"] is not None:
-        if "power" in red_knight["potion"]["effect"]:
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
-
-    # -------------------------------------------------------------------------------
-    # BATTLE:
-
-    # 1 Lancelot vs Mordred:
-    lancelot["hp"] -= mordred["power"] - lancelot["protection"]
-    mordred["hp"] -= lancelot["power"] - mordred["protection"]
-
-    # check if someone fell in battle
-    if lancelot["hp"] <= 0:
-        lancelot["hp"] = 0
-
-    if mordred["hp"] <= 0:
-        mordred["hp"] = 0
-
-    # 2 Arthur vs Red Knight:
-    arthur["hp"] -= red_knight["power"] - arthur["protection"]
-    red_knight["hp"] -= arthur["power"] - red_knight["protection"]
-
-    # check if someone fell in battle
-    if arthur["hp"] <= 0:
-        arthur["hp"] = 0
-
-    if red_knight["hp"] <= 0:
-        red_knight["hp"] = 0
-
-    # Return battle results:
-    return {
-        lancelot["name"]: lancelot["hp"],
-        arthur["name"]: arthur["hp"],
-        mordred["name"]: mordred["hp"],
-        red_knight["name"]: red_knight["hp"],
-    }
-
-
-print(battle(KNIGHTS))
+from app.arenas.camelot import Camelot
+from app.knights.knight import Knight
+
+
+def battle(knights_config: dict) -> dict:
+    battle_arena = Camelot()
+    [Knight(name=knight["name"],
+            hp=knight["hp"],
+            power=knight["power"],
+            armour=knight["armour"],
+            weapon=knight["weapon"],
+            potion=knight["potion"])
+        for knight in knights_config.values()]
+
+    battle_arena.knight_battle(Knight.knights_for_battle
+                               .get("Lancelot")
+                               .prepare_for_battle(),
+                               Knight.knights_for_battle
+                               .get("Mordred")
+                               .prepare_for_battle())
+    battle_arena.knight_battle(Knight.knights_for_battle
+                               .get("Arthur")
+                               .prepare_for_battle(),
+                               Knight.knights_for_battle
+                               .get("Red Knight")
+                               .prepare_for_battle())
+
+    return battle_arena.battles_results
```

Comment:
create additional variables, it has bad readability now

Reply:
I tired to add new variables, was still looking messy. Decided to adjust spacing and looks better to me now. What are your thoughts on it?